bin/adhoc-scripts/updatePileupObjects.py

```diff
@@ -147,6 +158,68 @@ def writePileupDocs(mspileupUrl, puDocs, handler, logger):
                 logger.critical(msg)
 
 
+def addTransitionRecords(handler, url, userDN, logger, dryrun):
+    """
+    Helper function to add and update in place transition record for existing MSPileup records
+    :param handler: HTTP handler to use
+    :param url: url to use (string)
+    :param userDN: string representing user DN
+    :param logger: logger to use
+    :param dryrun: option to run dry-run mode (boolean)
+    :return: nothing
+    """
+    puDocs = getPileupDocs(url, handler, logger)
+    logger.info("Found %d documents in MSPileup", len(puDocs))
+    for rec in puDocs:
+        if not rec.get("transition"):
+            tranRecord = {'containerFraction': 1.0,
+                          'customDID': rec['pileupName'],
+                          'updateTime': gmtimeSeconds(),
+                          'DN': userDN}
+            rec['transition'] = [tranRecord]
+            logger.info("New pileup document is: %s", rec)
+
+    if dryrun:
+        logger.info("documents are not written due to --dry-run option\n")
+    else:
+        # finally, update the pileup documents in the database
+        writePileupDocs(url, puDocs, handler, logger)
+        logger.info("documents are uploaded to MSPileup\n")
+
+
+def updatePileupRecords(handler, url, fin, logger, dryrun):
+    """
+    helper function to update pileup records
+    :param handler: HTTP handler to use
+    :param url: url to use (string)
+    :param fin: input file name to read from (string)
+    :param logger: logger object to use
+    :param dryrun: option to run dry-run mode (boolean)
+    :return: nothing
+    """
+    if os.path.exists(fin):
```

Comment:
I think the option parser needs to be updated and input file to become an optional command line argument.

Reply:
fixed.

Reply:
And yet you are checking if the file exists, regardless whether an input was passed in or not. This is definitely wrong.

Reply:
Why checking for file presence is wrong? What if I had a typo in provided argument. How about changing this to the following construct:
```
if fin:
    if os.path.exists(fin):
         # do action
    else:
         logger.error("Provided file %s does not exist", fin)
         os.exit(1)
else:
    # read docs from MongoDB
```

Reply:
Precisely my point, we should only check for file existence if a file/path name is provided.

Reply:
ok, cleared.